ops/testing.py

```diff
@@ -2500,38 +2617,54 @@ def make_dir(
             raise pebble.PathError(
                 'generic-file-error',
                 f'permissions not within 0o000 to 0o777: {permissions:#o}')
-        try:
-            self._fs.create_dir(
-                path, make_parents=make_parents, permissions=permissions,
-                user_id=user_id, user=user, group_id=group_id, group=group)
-        except FileNotFoundError as e:
-            # Parent directory doesn't exist and make_parents is False
+        self.__check_absolute_path(path)
+        dir_path = self._root / path[1:]
+        if not dir_path.parent.exists() and not make_parents:
             raise pebble.PathError(
-                'not-found', f'parent directory not found: {e.args[0]}')
+                'not-found', f'parent directory not found: {path}')
+        if not dir_path.parent.exists() and make_parents:
+            self.make_dir(
+                os.path.split(path)[0],
+                make_parents=True,
+                permissions=permissions,
+                user_id=user_id,
+                user=user,
+                group_id=group_id,
+                group=group)
+        try:
+            permissions = permissions if permissions else 0o755
```

Comment:
Similar to above, should we just skip the `os.chmod()` if permissions is None?

Reply:
In that case the test result will affected by the umask in developer's environment, which I would like to avoid.